### PR TITLE
NED-6376 update dlp ab test to be client based and not url based

### PR DESCRIPTION
**Pull request by haisleyb**

### CODE REVIEW

File: scripts/config/templates/nginx.conf.erb [hunk 1]
```diff
@@ -218,20 +218,32 @@ http {
...
+    }
+
+    # this proxies the request to the v1 prerendered dlp bucket folder if it is not a rendertron request and the URL is DLP-like
+    if ($dlp_ab_test_condition ~ "(*UTF8)^---false---\/locations\/[A-z\-\p{L}'%\/]*$") {
       proxy_pass https://s3.amazonaws.com/neighbor-build-${environment}/prerendered$request_uri;
       break;
     }
 
+    # this proxies the request to the v2 prerendered dlp bucket folder if it is not a rendertron request and the URL is DLP-like
+    if ($dlp_ab_test_condition ~ "(*UTF8)^---true---\/locations\/[A-z\-\p{L}'%\/]*$") {
```
Comment (birdsean) [inline scripts/config/templates/nginx.conf.erb:242]: do we care about keeping the ab test if someone is hitting an old url structure?

File: scripts/config/templates/nginx.conf.erb [hunk 2]
```diff
@@ -22,7 +22,7 @@ http {
 
   server_tokens off;
 
-  split_clients "${request_uri}_DLP_VERSION" $is_dlp_v2 { 
+  split_clients "${remote_addr}_DLP_VERSION" $is_dlp_v2 {
```
Comment (birdsean) [inline scripts/config/templates/nginx.conf.erb:25]: what if it's just a bunch of cloudfront servers hitting nginx?

File: scripts/config/templates/nginx.conf.erb (hunk 1 above)
Comment (haisleyb) [reply scripts/config/templates/nginx.conf.erb:242]: I don't think it hurts. It'll still fire the same events

File: scripts/config/templates/nginx.conf.erb (hunk 2 above)
Comment (haisleyb) [reply scripts/config/templates/nginx.conf.erb:25]: I don't understand your question. A client has to send a request to cloundfront that then forwards to nginx.

File: scripts/config/templates/nginx.conf.erb (hunk 1 above)
Comment (birdsean) [reply scripts/config/templates/nginx.conf.erb:242]: I guess I'm more concerned about the complexity of making everything backwards compatible over time. at some point we should just deprecate support for the old url and maybe that is now

File: scripts/config/templates/nginx.conf.erb (hunk 2 above)
Comment (haisleyb) [reply scripts/config/templates/nginx.conf.erb]: Updated to user x-forwarded-for which will contain chain of ips including real client ip

**Comment by prattprattpratt** [comment]
Old, closing

**Pull request closed without merging**